js/components/site-search.vue

```diff
@@ -0,0 +1,292 @@
+            cache: new Cache('site-search', sessionStorage),
+            groups: [
+                group('datasets', this._('Datasets')),
+                group('reuses', this._('Reuses')),
+                group('organizations', this._('Organizations'), 'organization'),
+                group('territory', this._('Territories'), 'territory'),
+            ],
+            minLength: 2,
+        }
+    },
+    computed: {
+        items() {
+            const items = [];
+            this.groups.forEach(group => {
+                if (group.items.length) {
+                    items.push(...group.items);
+                }
+            });
+            return items;
+        }
+    },
+    partials: {
+        default: `<div class="logo">
+            <img :src="item.image_url" class="avatar" width="30" height="30" alt="">
+            </div>
+            <p v-html="item.title | highlight query"></p>`,
+        organization: `<div class="logo"><img :src="item.image_url" class="avatar" width="30" height="30" alt=""></div>
+            <p v-html="item.name | highlight query"></p>
+            <small v-if="item.acronym" v-html="item.acronym | highlight query"></small>`,
+        territory: `<div class="logo">
+            <img :src="item.image_url" class="avatar" width="30" height="30" alt="">
+            </div>
+            <p v-html="item.title | highlight query"></p>
+            <small v-if="item.parent">{{ item.parent }}</small>`,
+    },
+    methods: {
+        reset() {
+            this.query = '';
+            this.current = -1;
+            this.show = false;
+            this.groups.forEach(group => group.items = []);
+        },
+        setActive(item) {
+            this.current = this.items.indexOf(item);
+        },
+        isActive(item) {
+            return this.current === this.items.indexOf(item);
+        },
+        hit(e) {
+            const item = this.items[this.current];
+            if (item) {
+                e.preventDefault()
+                window.location = item.page;
+            }
+        },
+        up() {
+            if (this.current >= 0)  {
+                this.current--;
+            } else {
+                this.current = this.items.length - 1;  // Cycle
+            }
+        },
+        down() {
+            if (this.current < this.items.length - 1) {
+                this.current++;
+            } else {
+                this.current = 0;  // Cycle
+            }
+        }
+    },
+    filters: {
+        highlight(value, phrase) {
+            return value.replace(new RegExp('('+phrase+')', 'gi'), '<strong>$1</strong>')
+        }
+    },
+    watch: {
+        query(value) {
+            const query = value.trim()
+            if (!query || query.length < this.minLength) {
+                this.reset()
+                return false
+            }
+
+            this.current = -1;
+            this.show = true;
+            
+            this.groups.forEach(group => {
```

Comment:
I tried but right now, territory suggest is too long to respond for the suggestions to show before submitting or typing new chars :/